### PR TITLE
Added docstring to clip_shp and add .rst documentation to clip_point

### PR DESCRIPTION
**Pull request by mlevis1**

### CODE REVIEW

File: earthpy/spatial.py [hunk 1]
```diff
@@ -161,16 +161,11 @@ def crop_image(raster, geoms, all_touched = True):
         Specifically the extent (shape elements) and transform properties are updated.
     """
 
-    # test that you have a list of a geodataframe
```
Comment (lwasser) [inline earthpy/spatial.py]: @mlevis1 i believe this was already done last week and merged. Did you update your fork before working on this? @jlpalomino  just giving you a heads  up!

**Comment by jlpalomino** [comment]
@mlevis1 Is it correct that this PR only addresses the first component of your issue (`Docstring documentation missing from clip_shp`)?  The second part of your issue is `RST documentation missing for clip_point`; are you going to commit additional changes to address that second component?

For the first component, please update that part of the title of the PR to be more specific: `Added docstring to clip_shp`. 

Good start on the docstring for clip_shp. Please make the following changes: 
1. Begin the docstring with double quotes """, rather than single quotes ''', and start the text on that same line. Elaborate on that first statement a bit: `"""A function to clip points, lines, polygon geometries based an input polygon geometry.`
2. Move the text after Returns to the description before Parameters. Also, update that text to be more specific about lines and polygons. Example:
`Depending on the geometry type,
input data will be clipped to the 
full extent of clip_obj using
either clip_points or clip_line_poly.`
3. Please add dashed lines under the words Parameters and Returns, as seen in this example from rasterio: https://github.com/mapbox/rasterio/blob/master/rasterio/mask.py
4. Update the Parameters and Returns to include the object type. Example: `shp: geopandas dataframe` (line 54). 
5. Remove the extra lines in the Parameters and Returns between the object type and the description. The description can also now include the lines removed to add the object type. Example: `vector layer to be clipped.` Example:
`shp: geopandas dataframe`
`vector layer (point, line, polygon) to be clipped to clip_obj` (Note: this line should be indented but this comment box will not let me show that).

File: earthpy/clip.py [hunk 2]
```diff
@@ -43,8 +44,31 @@ def clip_line_poly(shp, clip_obj):
...
+   extent of clip_obj using either clip_points or clip_line_poly.
+
+   If there are multiple polygons in object,
+   data will be clipped to the total boundary of
+   all polygons.
+
+   Parameters
+   ----------
+   shp : Geopandas dataframe
+         Dataset to be clipped.
```
Comment (jlpalomino) [inline earthpy/clip.py]: Please update line 62 to `Vector layer (point, line, polygon) to be clipped to clip_obj`.

File: earthpy/clip.py [hunk 3]
```diff
@@ -43,8 +44,31 @@ def clip_line_poly(shp, clip_obj):
...
-    '''
+   """A function to clip points, lines, polygon geometries based on an input
+   geometry.
+
+   Both layers must be the same Coordinate Reference System (CRS).
+
+   Depending on the geometry type, input data will be clipped to the full
+   extent of clip_obj using either clip_points or clip_line_poly.
+
+   If there are multiple polygons in object,
```
Comment (jlpalomino) [inline earthpy/clip.py]: Update line 55: `If there are multiple polygons in clip_obj,`

File: earthpy/clip.py [hunk 4]
```diff
@@ -43,8 +44,31 @@ def clip_line_poly(shp, clip_obj):
...
+   data will be clipped to the total boundary of
```
Comment (jlpalomino) [inline earthpy/clip.py]: Update line 56: `data from shp will be clipped to the total boundary of`

File: earthpy/clip.py [hunk 5]
```diff
@@ -43,8 +44,31 @@ def clip_line_poly(shp, clip_obj):
...
+   all polygons.
```
Comment (jlpalomino) [inline earthpy/clip.py]: Update line 57: `all polygons in clip_obj.`

**Comment by jlpalomino** [comment]
@mlevis1 Great job. A final review will be completed by @lwasser before the changes are merged. Thanks.

**Comment by lwasser** [comment]
This is an excellent pr @mlevis1 ! Thank you. I am merging it now! Congrats on your first contribution!! And thank you for helping to improve earthpy  .

**Pull request merged**